crypto/bio/b_print.c

```diff
@@ -842,19 +838,15 @@ int BIO_snprintf(char *buf, size_t n, const char *format, ...)
 int BIO_vsnprintf(char *buf, size_t n, const char *format, va_list args)
 {
     size_t retlen;
-    int truncated;
 
-    if(!_dopr(&buf, NULL, &n, &retlen, &truncated, format, args))
+    /*
+     * In case of truncation, return -1 like traditional snprintf.
+     * (Current drafts for ISO/IEC 9899 say snprintf should return the
+     * number of characters that would have been written, had the buffer
+     * been large enough.)
+     */
+    if(!_dopr(&buf, NULL, &n, &retlen, format, args))
```

Comment:
Having looked at this again, I'm not sure this is right. Here we are passing NULL for the dynamic buffer. When this ultimately gets passed down to doapr_outch the buffer expansion logic gets skipped if buffer == NULL. In fact it *always* returns 1 if buffer == NULL, so it does not fail in the event of truncation.